rockcraft/layers.py

```diff
@@ -45,14 +47,53 @@ def archive_layer(
     candidates = _gather_layer_paths(new_layer_dir, base_layer_dir)
     layer_paths = _merge_layer_paths(candidates)
 
-    with tarfile.open(temp_tar_file, mode="w") as tar_file:
-        # Iterate on sorted keys, so that the directories are always listed before
-        # any files that they contain (otherwise tools like Docker might choke on
-        # the layer tarball).
-        for arcname in sorted(layer_paths):
+    if not temp_tar_file.parent.exists():
+        temp_tar_file.parent.mkdir(parents=True)
+
+    layer_contents: list[Path] = []
+    with TemporaryDirectory() as tmpdir:
+        tmppath = Path(tmpdir)
+        # Just create an empty tar file if there's nothing to put in the layer
+        # GNU tar simply refuses to create an actual empty tar file though, so just use
+        # python's tarfile
+        if not layer_paths:
+            with tarfile.open(temp_tar_file, "w"):
+                pass
+            return
+
+        # Walk in reverse to avoid encountering not-yet created dirs
+        for arcname in sorted(layer_paths, reverse=True):
```

Comment:
question: I am not sure to understand why we now walk in reverse but before we walked in the "default" order. With `/a`, `/a/b`, `/a/b/c`, this implementation will try to add `/a/b/c` in the archive first, right? This might be problematic when extracted, right?